src/test/java/com/thealgorithms/datastructures/lists/RotateSinglyLinkedListsTest.java

```diff
@@ -0,0 +1,74 @@
+package com.thealgorithms.datastructures.lists;
+
+/**
+ * Test case of Rotate a list
+ * @author Bama Charan Chhandogi (https://github.com/BamaCharanChhandogi)
+ */
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+public class RotateSinglyLinkedListsTest {
+
+    @Test
+    public void testRotateRight() {
```

Comment:
Please split this test into multiple tests so that they all check separate cases

Reply:
done